### PR TITLE
[ao][sparsity] Bug Fix: Retain mask and config while replacing data in data sparsifier

### PR DESCRIPTION
**Pull request by macandro96**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82131
* #82130
* __->__ #82129

Bug: The config and mask were being recreated while replacing data on the data sparsifier.

Fix: Introduced an argument `reuse_mask` which when set `True` uses the old mask. If new config is not
specified, the data sparsifier by default uses the old config with the new data.
Also, added unit tests to check this bug.

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

### CODE REVIEW

File: torch/ao/sparsity/_experimental/data_sparsifier/base_data_sparsifier.py [hunk 1]
```diff
@@ -4,10 +4,14 @@
...
 from torch import nn
-import warnings
 import copy
 from ...sparsifier import utils
 from torch.nn.utils import parametrize
+import sys
+import warnings
+
+if not sys.warnoptions:
+    warnings.simplefilter("once")
```
Comment (z-a-f) [inline torch/ao/sparsity/_experimental/data_sparsifier/base_data_sparsifier.py:14]: nit: Can you add a comment here why the warnings are filtered?

File: torch/ao/sparsity/_experimental/data_sparsifier/base_data_sparsifier.py [hunk 2]
```diff
@@ -74,8 +78,15 @@ def _extract_weight(self, data):
         elif type(data) in EMBEDDING_TYPES:
             return data.weight
 
-    def add_data(self, name: str, data, **config):
-        r""" Configures and parametrizes the internal container model with name and data
+    def add_data(self, name: str, data, use_old_mask=True, **config):
```
Comment (z-a-f) [inline torch/ao/sparsity/_experimental/data_sparsifier/base_data_sparsifier.py:81]: Can we rename the `use_old_mask` into `reuse_mask` (or something like that)? imho, it's avoids the questions, such as "what do you mean by old?"

**Review by z-a-f — APPROVED** [review]
